backend/src/main/java/gov/cdc/usds/simplereport/idp/authentication/LiveOktaAuthentication.java

```diff
@@ -0,0 +1,138 @@
+package gov.cdc.usds.simplereport.idp.authentication;
+
+import com.okta.sdk.authc.credentials.TokenClientCredentials;
+import com.okta.sdk.client.Client;
+import com.okta.sdk.client.Clients;
+import com.okta.sdk.resource.ResourceException;
+import com.okta.sdk.resource.user.PasswordCredential;
+import com.okta.sdk.resource.user.RecoveryQuestionCredential;
+import com.okta.sdk.resource.user.User;
+import com.okta.sdk.resource.user.UserCredentials;
+import com.okta.spring.boot.sdk.config.OktaClientProperties;
+import gov.cdc.usds.simplereport.api.model.errors.InvalidActivationLinkException;
+import gov.cdc.usds.simplereport.api.model.errors.OktaAuthenticationFailureException;
+import gov.cdc.usds.simplereport.config.BeanProfiles;
+import java.util.List;
+import org.json.JSONObject;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Created by emmastephenson on 4/28/21
+ *
+ * <p>Handles all Okta-related authenticaton.
+ */
+@Profile("!" + BeanProfiles.NO_OKTA_AUTH)
+@Service
+public class LiveOktaAuthentication implements OktaAuthentication {
+  private Client _client;
+  private String _apiToken;
+  private String _orgUrl;
+
+  public LiveOktaAuthentication(OktaClientProperties oktaClientProperties) {
+    _client =
```

Comment:
Yeah, I thought about this too. The original reason I created this file is that I was going to use the `auth` SDK, and I thought it would make more sense to separate the SDK usage into different files.

Now of course things have changed and this does mostly use the management SDK, but I still lean towards keeping them separate because as this Okta auth project keeps going, we will have use cases where we need to use the auth SDK. (i.e. I don't think logging in is supported in the management SDK). And we definitely could just move it all to the repository, but I also worry about the separation of concerns/business logic. I'm not familiar enough with the client caching to know the details there, but I agree that for now the low number of requests probably means it won't matter too much. Maybe we can revisit later to see if we should move to a shared cache?